src/page/Post/Post.tsx

```diff
@@ -13,37 +12,39 @@ import { usePosts } from '../../features/posts/usePosts'
 import { Link } from 'react-router-dom';
 
 export const Post = () => {
-
-    const { posts, onLikePost, onDislikePost, onFavoritePost } = usePosts()
-
+    
+    const { getInfoOnePost, onePost, onLikePost, onDislikePost, onFavoritePost } = usePosts()
+    
     const { id } = useParams();
 
+    const infoPost = getInfoOnePost(Number(id)-1);
```

Comment:
Всегда получай информацию о посте по id и не преобразовывай его
```suggestion
    const infoPost = getInfoOnePost(id);
```